ocaml-backend/src/main/java/org/kframework/backend/ocaml/DefinitionToOcaml.java

```diff
@@ -477,11 +480,34 @@ private void matchPatternAndPrint(StringBuilder sb) {
         sb.append("let _ = (try print_subst_binary subst (try_match res res (-1)) with Stuck c -> output_string subst \"\\x00\\x00\\x00\\x00\")\n");
     }
 
+    private void generateMain(StringBuilder sb) {
+        indentDepth = 0;
+        indented(sb).append("let main =\n");
+        indentDepth++;
+    }
+
+    private void callMain(StringBuilder sb) {
+        indentDepth = 0;
+        indented(sb).append("let _ = main");
+    }
+
     private void runAndPrint(int depth, StringBuilder sb) {
-        sb.append("let res, steps = ");
+        indented(sb).append("try (\n");
+        indentDepth++;
+        indented(sb).append("let res, steps = ");
         constructRun(sb, String.valueOf(depth));
+        sb.append("in\n");
+        indented(sb).append("output_string out (");
+        sb.append("(string_of_int steps) ^ \"\\n\"");
+        sb.append(" ^ (string_of_int 1) ^ \"\\n\"");
+        sb.append(" ^ print_k_binary(res))\n");
+        indentDepth = 1;
+        indented(sb).append(") with (FunctionEvalFailed' (n,stacktrace)) -> (output_string out (");
+        sb.append("(string_of_int n) ^ \"\\n\"");
```

Comment:
This code is very difficult to make out, but it seems suspicious to me. I thought the whole idea was that we were just going to be returning a K term? why does this need to be modified at all?

Reply:
This way we write 3 things to the output file:
1. number of steps
2. whether everything was ok or evaluation failed
3. the resulting term(s)

Concerning (2), I prefer to have such a such flag to the other options (either detecting a `List` term, or counting the number of terms), because the othre options are about having some special values. This way we may treat the term(s) opaquely. E.g. we do not have to rely on the assumption that a valid result is never a `List`.

Concerning (3), if we returned someting like `List(ListItem(k1), ..., ListItem(kn))`, the Java code would have to either print the term as-is, resulting in something like
```
ListItem ( isNice ( 0 ) )
ListItem ( isPretty ( 0 ) )
ListItem ( isBeautiful ( 0 ) )
ListItem ( <T>
  <k>
    9 / 0
  </k>
</T> )
```
, or somehow look into the `List` - which I am not sure how exactly to do it.

If your concern is that it is not easy to se what to generated ocaml code will do - I will clean it up.